mlflow/pyfunc/__init__.py

```diff
@@ -502,7 +502,11 @@ def load_model(
 
     _add_code_from_conf_to_system_path(local_path, conf, code_key=CODE)
     data_path = os.path.join(local_path, conf[DATA]) if (DATA in conf) else local_path
-    model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
+    loader_module = conf[MAIN]
+    if loader_module in ["mlflow.tensorflow", "mlflow.keras"]:
+        model_impl = mlflow.tensorflow._load_pyfunc(data_path)
+    else:
+        model_impl = importlib.import_module(loader_module)._load_pyfunc(data_path)
```

Comment:
I removed the model_type arg, but we still have to add the branch here,
because we need to map both "mlflow.tensorflow", "mlflow.keras" to `mlflow.tensorflow` module. @dbczumar

Reply:
The change https://github.com/mlflow/mlflow/pull/6530/commits/547638c03b234ba416516e9e677b54c3b1e2bc0b does not work,
because `importlib.import_module(loader_module)._load_pyfunc(data_path)` need to import "mlflow.keras" module, we need a real module there,

so I use approach https://github.com/mlflow/mlflow/pull/6530/commits/ac45abb1b820117c6e5c3ae04b51ac50287e9169 instead.

Reply:
Sounds good! Can we also update your new `mlflow.keras` module so that `mlflow.keras.log_model = mlflow.tensorflow.log_model`, `mlflow.keras.save_model = mlflow.tensorflow.save_model`, `mlflow.keras.autolog = mlflow.tensorflow.autolog`, etc?

Reply:
Updated :)

Reply:
> The change https://github.com/mlflow/mlflow/commit/547638c03b234ba416516e9e677b54c3b1e2bc0b does not work,
because importlib.import_module(loader_module)._load_pyfunc(data_path) need to import "mlflow.keras" module, we need a real module there,


Doesn't this work?

```diff
-     model_impl = importlib.import_module(loader_module)._load_pyfunc(data_path)
+     loader_module = "mlflow.tensorflow" if loader_module == "mlflow.keras" else loader_module
+     model_impl = importlib.import_module(loader_module)._load_pyfunc(data_path)

```

Reply:
```
-     model_impl = importlib.import_module(loader_module)._load_pyfunc(data_path)
+     loader_module = "mlflow.tensorflow" if loader_module == "mlflow.keras" else loader_module
+     model_impl = importlib.import_module(loader_module)._load_pyfunc(data_path)
```
This works, but @dbczumar does not like this approach.

Reply:
I'd rather not special case flavor loading logic in `mlflow.pyfunc`.